src/decl.h

```diff
@@ -55,6 +55,85 @@ class Declarator;
 #define TYPEQUAL_VECTORCALL (1 << 10)
 #define TYPEQUAL_REGCALL (1 << 11)
 
+enum AttrArgKind { ATTR_ARG_UINT32, ATTR_ARG_STRING, ATTR_ARG_UNKNOWN };
+
+/** @brief Representation of an argument to an attribute.
+
+    Attribute can have an argument, which can be either an integer or a string.
+    This class is used to represent those arguments.
+ */
+class AttrArgument {
+  public:
+    AttrArgument();
+    AttrArgument(int64_t i);
+    AttrArgument(const std::string &s);
+
+    void Print() const;
+
+    AttrArgKind kind;
+    int64_t intVal;
+    std::string stringVal;
+};
+
+/** @brief Representation of a single attribute.
+
+    Attributes are used to provide additional information about a
+    declaration that isn't captured by the basic type, storage class, etc.
+    Its semantics are contained in the name and the argument, if any.
+ */
+class Attribute {
+  public:
+    Attribute(const std::string &n);
+    Attribute(const std::string &n, AttrArgument a);
+    Attribute(const Attribute &a);
+
+    /** Returns true if the attribute is known/supported false otherwise. */
+    bool IsKnownAttribute() const;
+
+    void Print() const;
```

Comment:
No tests with `--ast-dump` were added / modified. Are the attributes printed / printed correctly with `--ast-dump`?

Do we need `void Print(&Indent)` version?

Reply:
I have added test `ast-dump-attrs.ispc` but I don't see any traces of `DeclSpecs` and `Declarator` in AST dumps.

Reply:
Looking at the test, I assume these attributes are printed as part of AST dump. If so, I'm good with it.